Code/Common/include/sitkObjectOwnedBase.h

```diff
@@ -0,0 +1,115 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef sitkObjectOwnedBase_h
+#define sitkObjectOwnedBase_h
+
+
+#include "sitkCommon.h"
+#include "sitkNonCopyable.h"
+
+#include <map>
+#include <functional>
+
+
+namespace itk
+{
+
+class Object;
+
+namespace simple
+{
+
+/** \class ObjectOwnedBase
+ * \brief  An abstract base class to connect this object with the lifetime of another.
+ *
+ * This class maintains a list of Objects and callbacks to call when this object is deleted.
+ *
+ * Optionally, "OwnedByObjects" property can be enabled so that when the last referenced object is removed this object
+ * will be deleted.
+ *
+ *
+ */
+class SITKCommon_EXPORT ObjectOwnedBase
+: public NonCopyable
+{
+public:
+  virtual ~ObjectOwnedBase() = 0;
```

Comment:
After an excessive amount of trial, error, and simplifying the R SWIG error, it appears that the `= 0` is causing problems with R SWIG. Perhaps this was causing issue with the Python wrapping I encountered earlier too.